csharp_dotnetcore/17.Proactive-Messages/State/JobState.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder;
+
+/// <summary>A <see cref="BotState"/> for managing bot state for "bot jobs".</summary>
+/// <remarks>Independent from both <see cref="UserState"/> and <see cref="ConversationState"/> because
+/// the process of running the jobs and notifying the user interacts with the
+/// bot as a distinct user on a separate conversation.</remarks>
+public class JobState : BotState
+{
```

Comment:
Is there a way to refactor this sample to NOT need this? 

It's a huge concept - and an good one - but it's a big concept and not really done justice in this sample. 

... it's also not clear that this is a Middlware component (BotState needs to be renamed)

Reply:
Since BotState is abstract, and we don't want to use ConversationState or UserState, I didn't see a way around this. The only other option is to read and write directly to storage and skip having a state management object completely.